rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -212,116 +263,121 @@ def configure_optimizers(self) -> ParamOptimizerPairs:
         parameters. To support both tf and torch, we must explicitly return the
         parameters as the first element of the tuple regardless of whether those
         exist in the optimizer objects or not. This method is called once at
-        initialization and everytime a new sub-module is added to the module.
+        initialization.
 
         Returns:
             A list of tuples (parameters, optimizer), where parameters is a list of
             model parameters and optimizer is a deep learning optimizer.
+        """
+
+    @abc.abstractmethod
+    def compute_gradients(self, loss: Mapping[str, Any]) -> ParamDictType:
+        """Computes the gradients based on the loss.
 
+        Args:
+            loss: The computed loss dict. It should include the key
+                `self.TOTAL_LOSS_KEY` that contains the total loss.
+        Returns:
+            The gradients in teh same format as self._params.
         """
 
-    def compute_loss(
```

Comment:
All of these red / greens are just me re-ordering the api locations. Don't panic, I am not removing all the stuff we wrote :)